crates/ruma-macros/src/lib.rs

```diff
@@ -10,17 +10,25 @@ use proc_macro::TokenStream;
 use proc_macro2 as pm2;
 use proc_macro_crate::{crate_name, FoundCrate};
 use quote::{format_ident, quote};
+use ruma_identifiers_validation::{
+    device_key_id, event_id, key_id, mxc_uri, room_alias_id, room_id, room_version_id, server_name,
+    user_id,
+};
 use syn::{parse_macro_input, DeriveInput};
 
-use self::events::{
-    event::expand_event,
-    event_content::expand_event_content,
-    event_enum::{expand_event_enums, expand_from_impls_derived},
-    event_parse::EventEnumInput,
-    event_type::expand_event_type_enum,
+use self::{
+    events::{
+        event::expand_event,
+        event_content::expand_event_content,
+        event_enum::{expand_event_enums, expand_from_impls_derived},
+        event_parse::EventEnumInput,
+        event_type::expand_event_type_enum,
+    },
+    identifiers::IdentifierInput,
 };
 
 mod events;
+mod identifiers;
```

Comment:
I wonder, could this module be public and act as the place that exports the ID proc-macros? (maybe not, I'm not sure on the `#[proc_macro]` rules right now)

Reply:
I would prefer it also but it seems not. The proc macros are required to be at the root of the crate.